src/cli/tls_stream_http_server.cpp

```diff
@@ -0,0 +1,556 @@
+/*
+* (C) 2014,2015,2017,2019 Jack Lloyd
+* (C) 2016 Matthias Gierlings
+* (C) 2019 Hannes Rantzsch
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*
+*
+* This code is based on the Beast Example: HTTP SSL server, asynchronous.
+*
+* Copyright (c) 2016-2019 Vinnie Falco (vinnie dot falco at gmail dot com)
```

Comment:
This is a simple file server for the `botan-cli` based on the Beast example "HTTP SSL server, asynchronous". @randombit @vinniefalco can you confirm that including the Beast copyright notice is OK with both the license requirements of Botan and Beast?

Reply:
Looks fine to me, and I'm glad that the code is useful to you.

Reply:
Alas we cannot take this. There are many organizations who carefully track exactly which license(s) are being used in their open source dependencies, I am in fairly regular contact with many large industrials, etc on this topic, and combining multiple licenses is a complication and uncertainty for them that I don't want to bring in.

For similar reasons Botan is unable to make use of MIT or BSD-3 licensed code even though of course MIT/BSD-2/BSD-3/BSL are all basically the same thing.

[One could argue that few of these orgs are paying me, so I shouldn't care too much about their concerns, but the main reason I maintain this lib at all is so something exists which can be usable in large commercial projects that is not named OpenSSL, and anything which deters such usage makes the work less satisfying for me on a personal level]

Sorry :/

Reply:
I understand.

> MIT/BSD-2/BSD-3/BSL are all basically the same thing.

Hmm, no not at all there is a world of difference. BSD-2 requires attribution in binaries while the BSL does not. BSD-2 is less permissive.

Reply:
> combining multiple licenses is a complication and uncertainty for them that I don't want to bring in

That makes sense. To my understanding though, we weren't releasing Botan or any part of it under BSL license. The code makes use of BSL licensed code, maintaining the copyright notice with the source code, as required. But it is still licensed BSD-2, like the rest of Botan, isn't it?

Did I misunderstand this, or can we solve this by a clearer wording?